tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -86,7 +87,235 @@ TEST( WaterThermalTankData, GetDeadBandTemp )
+		"    ,                        !- Maximum Sensible Heating Capacity {W}",
+		"    NoLimit,                 !- Cooling Limit",
+		"    autosize,                !- Maximum Cooling Air Flow Rate {m3/s}",
+		"    ,                        !- Maximum Total Cooling Capacity {W}",
+		"    ,                        !- Heating Availability Schedule Name",
+		"    ,                        !- Cooling Availability Schedule Name",
+		"    ConstantSupplyHumidityRatio,  !- Dehumidification Control Type",
+		"    ,                        !- Cooling Sensible Heat Ratio {dimensionless}",
+		"    ConstantSupplyHumidityRatio,  !- Humidification Control Type",
+		"    ,                        !- Design Specification Outdoor Air Object Name",
+		"    ,                        !- Outdoor Air Inlet Node Name",
+		"    ,                        !- Demand Controlled Ventilation Type",
+		"    ,                        !- Outdoor Air Economizer Type",
+		"    ,                        !- Heat Recovery Type",
+		"    ,                        !- Sensible Heat Recovery Effectiveness {dimensionless}",
+		"    ;                        !- Latent Heat Recovery Effectiveness {dimensionless}",
+		"  ZoneHVAC:EquipmentConnections,",
+		"    SPACE4-1,                !- Zone Name",
+		"    SPACE4-1 Eq,             !- Zone Conditioning Equipment List Name",
+		"    SPACE4-1 In Nodes,       !- Zone Air Inlet Node or NodeList Name",
+		"    SPACE4-1 Out Nodes,      !- Zone Air Exhaust Node or NodeList Name",
+		"    SPACE4-1 Node,           !- Zone Air Node Name",
+		"    SPACE4-1 Out Node;       !- Zone Return Air Node Name",
+		"  ZoneHVAC:EquipmentList,",
+		"    SPACE4-1 Eq,             !- Name",
+		"    ZoneHVAC:IdealLoadsAirSystem,  !- Zone Equipment 1 Object Type",
+		"    SPACE4-1 AirSys,            !- Zone Equipment 1 Name",
+		"    1,                       !- Zone Equipment 1 Cooling Sequence",
+		"    1,                       !- Zone Equipment 1 Heating or No-Load Sequence",
+		"    WaterHeater:HeatPump:PumpedCondenser,    !- Zone Equipment 2 Object Type",
+		"    Zone4HeatPumpWaterHeater,!- Zone Equipment 2 Name",
+		"    2,                       !- Zone Equipment 2 Cooling Sequence",
+		"    2;                       !- Zone Equipment 2 Heating or No-Load Sequence",
+		"  NodeList,",
+		"    SPACE4-1 In Nodes,       !- Name",
+		"    SPACE4-1 In Node,        !- Node 1 Name",
+		"    Zone4AirInletNode,       !- Node 2 Name",
+		"    IdealLoadInNode;       !- Node 2 Name",
+		"  NodeList,",
+		"    SPACE4-1 Out Nodes,      !- Name",
+		"    Zone4AirOutletNode,      !- Node 1 Name",
+		"    IdealLoadOutNode;      !- Node 1 Name",
+	});
+
+	ASSERT_FALSE( process_idf( idf_objects ) );
+
+	bool ErrorsFound = false;
+	HeatBalanceManager::GetZoneData( ErrorsFound );
+	ASSERT_FALSE( ErrorsFound );
+	EXPECT_FALSE( WaterThermalTanks::GetWaterThermalTankInput() );
+}
+
+TEST_F( HVACFixture, HPWHWrappedDummyNodeConfig )
```

Comment:
Merging in this other unit test from develop caused me some problems. It's failing now because the state isn't being cleared for water heaters. @EnergyArchmage  is there a implementation of `clear_state` for `WaterThermalTanks` in that branch you're maintaining? I would have looked myself, but @Myoldmopar didn't mention the branch name in the email so I thought it would be easier to just ask you.